anchor/network/src/discovery.rs

```diff
@@ -0,0 +1,470 @@
+        &mut self,
+        _connection_id: ConnectionId,
+        _peer: PeerId,
+        _addr: &Multiaddr,
+        _role_override: Endpoint,
+        _port_use: PortUse,
+    ) -> Result<THandler<Self>, ConnectionDenied> {
+        Ok(ConnectionHandler)
+    }
+
+    fn on_swarm_event(&mut self, event: FromSwarm) {
+        match event {
+            FromSwarm::ConnectionEstablished(c) => {
+                log::debug!("Connection established: {:?}", c);
+            }
+            _ => {
+                // TODO handle other events
+            }
+        }
+    }
+
+    fn on_connection_handler_event(
+        &mut self,
+        _peer_id: PeerId,
+        _connection_id: ConnectionId,
+        _event: THandlerOutEvent<Self>,
+    ) {
+    }
+
+    fn poll(
+        &mut self,
+        cx: &mut Context<'_>,
+    ) -> Poll<ToSwarm<Self::ToSwarm, THandlerInEvent<Self>>> {
+        if !self.started {
+            return Poll::Pending;
+        }
+
+        // Process the query queue
+        //self.process_queue();
+
+        // Drive the queries and return any results from completed queries
+        if let Some(peers) = self.poll_queries(cx) {
+            // return the result to the peer manager
+            return Poll::Ready(ToSwarm::GenerateEvent(DiscoveredPeers { peers }));
+        }
+        Poll::Pending
+    }
+}
+
+/// Builds a anchor ENR given a `network::Config`.
+pub fn build_enr(enr_key: &CombinedKey, config: &Config) -> Result<Enr, String> {
+    let mut builder = discv5::enr::Enr::builder();
+    let (maybe_ipv4_address, maybe_ipv6_address) = &config.enr_address;
+
+    if let Some(ip) = maybe_ipv4_address {
+        builder.ip4(*ip);
+    }
+
+    if let Some(ip) = maybe_ipv6_address {
+        builder.ip6(*ip);
+    }
+
+    if let Some(udp4_port) = config.enr_udp4_port {
+        builder.udp4(udp4_port.get());
+    }
+
+    if let Some(udp6_port) = config.enr_udp6_port {
+        builder.udp6(udp6_port.get());
+    }
+
+    // Add QUIC fields to the ENR.
+    // Since QUIC is used as an alternative transport for the libp2p protocols,
+    // the related fields should only be added when both QUIC and libp2p are enabled
+    if !config.disable_quic_support {
+        // If we are listening on ipv4, add the quic ipv4 port.
+        if let Some(quic4_port) = config.enr_quic4_port.or_else(|| {
+            config
+                .listen_addresses
+                .v4()
+                .and_then(|v4_addr| v4_addr.quic_port.try_into().ok())
+        }) {
+            builder.add_value(QUIC_ENR_KEY, &quic4_port.get());
+        }
+
+        // If we are listening on ipv6, add the quic ipv6 port.
+        if let Some(quic6_port) = config.enr_quic6_port.or_else(|| {
+            config
+                .listen_addresses
+                .v6()
+                .and_then(|v6_addr| v6_addr.quic_port.try_into().ok())
+        }) {
+            builder.add_value(QUIC6_ENR_KEY, &quic6_port.get());
+        }
+    }
```

Comment:
the spec only mentions TCP as a transport, and the `go` `ssv` client doesn't seem to support it, do we plan to support it nonetheless?

Reply:
We could support it for communication between rust nodes.

Reply:
Yeah, lets support it, and hopefully they follow suit